### ISSUE TITLE
ruTorrent v5.1 - Beta (Please reply here for feedback)

### ISSUE THREAD

**Issue by stickz** [Novik/ruTorrent#2721]
### What would you like changed about the web client?

I would like the legend not to overlap with text when scrolling on mobile.

### Describe the solution you'd like and propose possible alternatives.

Implementing appropriate CSS style changes to avoid the undesired behavior.

### Provide information and resources about the environment hosting the web client.

Server: ruTorrent v5.0-Beta 1 on Ubuntu 24.04 LTS 
Web Client: Google Chrome 129.0.6668.59
Installation Method: zip source from version release

### Additional context

I believe @jevenski is able to resolve this problem. Here is a screenshot to provide additional context.

![ruTorrent_v5_legend_issue](https://github.com/user-attachments/assets/fa4190f7-83dd-4a47-8b7d-0a40344ff531)

**Comment by jevenski** [comment]
Can we preserve the sticky feature of the legend text, that sticks to the viewport until the cobtaining box is scrolled completely outside the viewport? I like this one, as it's cool and helpful for long fieldsets blocks. We can define the background color of the legend text to prevent the overlapping.

**Comment by jevenski** [comment]
Yes, we can have the elements inside the group. We just add a non-transparent background to the legend text to cover the elements underneath to prevent overlapping. I've created a new pull request here (with a screenshot inside):

https://github.com/Novik/ruTorrent/pull/2722

**Comment by Airman8** [comment]
### What would you like changed about the web client?

1: On mobile, the side menu accessed by the "..." button takes up the entire width of the viewport, resulting in there being no room beside the menu to touch and close out of it. If I switch my phone to landscape mode and then back to portrait mode again, the menu width is correct (the same as on my PC). Refreshing the page after switching orientation results in the menu again expanding to the full viewport width.

2: Another minor thing is the tabs in the lower torrent info section (General, Files, Trackers, etc.). When I press a tab, it slowly expands the tab bar vertically and pushes down the content of the tab below. It also results in the tab quickly snapping up and then sagging down again every time I press a different tab.

### Describe the solution you'd like and propose possible alternatives.

1: Probably needs a CSS fix to force the menu not to expand all the way, or perhaps a way to properly detect the media width on first load? Otherwise, a button could be added to the left of the "ruTorrent v5.B.1" label to close the side menu. Another option would be to shift the vertical position of the side menu down so as not to cover the button on the main page.

2: I think the tab pop-out behaviour would be okay if it were faster and if the tab bar had a fixed height; however, I personally would prefer no transition at all, since the content of the tab loads immediately without transition anyway.

### Provide information and resources about the environment hosting the web client

OS: Raspberry Pi OS (Debian Bookworm)
Server: nginx
Installation method: tarball source from version release
Web browser:
 - Chrome for Android version 129.0.6668.71
 - Firefox for Android version 130.0.1
Device: Samsung Galaxy S9

Thank you for all the effort on this UI rewrite. It's looking

**Comment by jevenski** [comment]
Hi, @Airman8 , thank you for testing and the all feedback!

> 1: On mobile, the side menu accessed by the "..." button takes up the entire width of the viewport, resulting in there being no room beside the menu to touch and close out of it. If I switch my phone to landscape mode and then back to portrait mode again, the menu width is correct (the same as on my PC). Refreshing the page after switching orientation results in the menu again expanding to the full viewport width.

There is a close button on the top-right corner of the panel. I realize that it might not be easy enough to spot, especially in dark themes. We'll adjust the color schemes if that's the problem. As for the side panel width, the current version won't set the width of the side panel for mobile screens (< 768px), so it would take the default value of the Bootstrap framework. We can adjust the process to set the width of the side panel for any screen size.

> 2: Another minor thing is the tabs in the lower torrent info section (General, Files, Trackers, etc.). When I press a tab, it slowly expands the tab bar vertically and pushes down the content of the tab below. It also results in the tab quickly snapping up and then sagging down again every time I press a different tab.

I didn't see the slow transition on the tab list and the tab detail page. Have you tried disabling the "iPad" plugin? This plugin simulates mouse events on touch events, and causes strange behaviors. It's going to be remove in the final stable release. For now, just disabling it should be fine.

**Comment by stickz** [comment]
@Airman8 Thank you for your feedback. We have addressed it and tagged [ruTorrent v5.0-Beta-2](https://github.com/Novik/ruTorrent/releases/tag/v5.0-beta2) with 10 bug fixes. Would you like to test it (with the ipad plugin deleted) and let us know what you think?

**Comment by Airman8** [comment]
> @Airman8 Thank you for your feedback. We have addressed it and tagged [ruTorrent v5.0-Beta-2](https://github.com/Novik/ruTorrent/releases/tag/v5.0-beta2) with 10 bug fixes. Would you like to test it (with the ipad plugin deleted) and let us know what you think?

@stickz Sure, I installed v5.0-beta2 and have been looking at all the themes.

> There is a close button on the top-right corner of the panel. I realize that it might not be easy enough to spot, especially in dark themes. We'll adjust the color schemes if that's the problem.

@jevenski I realize I had neglected to mention that I was using the MaterialDesign theme. I edited my comment for brevity and accidentally cut that part out - my apologies. I turned up the brightness on my phone and did see the button. 

> I didn't see the slow transition on the tab list and the tab detail page. Have you tried disabling the "iPad" plugin? This plugin simulates mouse events on touch events, and causes strange behaviors. It's going to be remove in the final stable release. For now, just disabling it should be fine.

I checked that the ipad plugin was disabled in my previous installation, and it has been removed in v5.0-beta2 as well.

For v5.0-beta2, I tried all the other themes on both desktop and mobile. I noticed that this particular issue with the tab bar growing downward is only happening with the MaterialDesign theme.

I also found that clicking on certain tabs in all the other themes causes the whole bottom section to grow vertically by 1 or 2 pixels. The tabs that do this the most often are the history and plugins tabs, but it happens with others depending on the theme. (Incidentally, this doesn't happen in the MaterialDesign theme.) I thought that it might be caused by the content of the active tab having x-overflow content, which could cause the scrollbar to appear and maybe bump the tab upward. But it happens even if I make the various fields in each tab narrower so that everything fits on the screen, so I'm not sure what's really making it happen. This isn't a big usability issue or anything. It's just a quirky UI thing where I can see the tab bar moving slightly up and down as I click through the tabs.

As for the X button in the side menu, it does look like just a matter of dark themes vs light themes. The X is always black, so it doesn't show up well in dark themes. Is it possible to get the colour of the X to match the colour of the other menu buttons? (for example, grey in the dark theme and blue in the MaterialDesign theme)

In the "DarkBetter" theme, the triangle icon on the side menu which indicates whether sections are visible or folded gets bigger for folded sections. This does not happen in the "Dark" theme though.

I was messing around in the right-click shortcut menu to see how it felt to navigate, and I accidentally removed a torrent because I touched "remove" when I meant to press "set channel". This could be a fatal "off-by-one error" if someone were to accidentally remove a whole label or tracker worth of torrents. In case anyone else has fat fingers like me, would it be possible to increase the vertical padding of the menu items on the mobile layout?

Looking forward to the full release!

**Comment by koblack** [comment]
1. js error when trying to add or remove icon for label

```
webui.js?v=5b2:2461 Uncaught TypeError: catlist.refreshTorrentLabelTree is not a function from TypeError: catlist.refreshTorrentLabelTree is not a function
    at request.onloadend (<anonymous>:7119:15)
```

2. plugin: rssurlrewrite: Unable to add new rules. The OK button doesn't seem to work. Unable to save changes.

![image](https://github.com/user-attachments/assets/f68acb68-ef89-487a-bf15-8c9ed69ece6a)

3. plugin: rss: Unable to add new RSS group. The OK button doesn't seem to work. Unable to save changes.

![image](https://github.com/user-attachments/assets/87b50447-e8d8-48dc-863a-dd24aa4575a8)


4. Allow to make the bottom panel bigger. This is especially useful when browsing torrents containing thousands of files. Version v4.3.6 allowed to enlarge it to the maximum.

![image](https://github.com/user-attachments/assets/1b4a7e60-7514-4c06-bc5c-c470d02ec272)
![image](https://github.com/user-attachments/assets/f8baf220-3846-412a-92a5-0c6e9f8bfded)

5. On mobile when you open a window from the side panel you can't enter text. For example, try changing the name of a view.

![Screenshot_20241004_200857_Chrome](https://github.com/user-attachments/assets/7eac6502-0e65-46d3-92a0-5eb7740391ce)

6. plugin: screenshots: When you turn on the slide show, the PLAY button is still displayed instead of STOP.

![image](https://github.com/user-attachments/assets/6cb7beee-da28-4f55-9d5a-f17630c22dc7)
![image](https://github.com/user-attachments/assets/d7e34298-d358-4341-b522-8eb4213f6712)

**Comment by jevenski** [comment]
> 1: On mobile, the side menu accessed by the "..." button takes up the entire width of the viewport, resulting in there being no room beside the menu to touch and close out of it. If I switch my phone to landscape mode and then back to portrait mode again, the menu width is correct (the same as on my PC). Refreshing the page after switching orientation results in the menu again expanding to the full viewport width.

Hi, @Airman8 , I adjusted the resize function of side panel to skip resizing on mobile, Now the side panel should always be of default width of Bootstrap's offcanvas panel.

Hi, @koblack , here's some fixes according to your feedback. The others were probably introduced by some earlier commits, and I'm still on the way of getting familiar with them, and not 100% sure how to fix them properly. Perhaps someone can fix those before I figure out how those features work.

>     2. plugin: rssurlrewrite: Unable to add new rules. The OK button doesn't seem to work. Unable to save changes.
>     3. plugin: rss: Unable to add new RSS group. The OK button doesn't seem to work. Unable to save changes.
>     6. plugin: screenshots: When you turn on the slide show, the PLAY button is still displayed instead of STOP.
These three bugs were fixed.

>     4. Allow to make the bottom panel bigger. This is especially useful when browsing torrents containing thousands of files. Version v4.3.6 allowed to enlarge it to the maximum.
Options were added to the General UI settings to allow setting the min width of the side panel and the min height of the list table. You can set the values as per your need.


Related PR: https://github.com/Novik/ruTorrent/pull/2723

**Comment by jevenski** [comment]
> For v5.0-beta2, I tried all the other themes on both desktop and mobile. I noticed that this particular issue with the tab bar growing downward is only happening with the MaterialDesign theme.
> 
> 
> 
> I also found that clicking on certain tabs in all the other themes causes the whole bottom section to grow vertically by 1 or 2 pixels. The tabs that do this the most often are the history and plugins tabs, but it happens with others depending on the theme. (Incidentally, this doesn't happen in the MaterialDesign theme.) I thought that it might be caused by the content of the active tab having x-overflow content, which could cause the scrollbar to appear and maybe bump the tab upward. But it happens even if I make the various fields in each tab narrower so that everything fits on the screen, so I'm not sure what's really making it happen. This isn't a big usability issue or anything. It's just a quirky UI thing where I can see the tab bar moving slightly up and down as I click through the tabs.

These minor issues are less urgent than the other issues below, but could take longer time to diagnose, so I think we can let them go now and come back to them after the stable release and fix them in v5.1.

> As for the X button in the side menu, it does look like just a matter of dark themes vs light themes. The X is always black, so it doesn't show up well in dark themes. Is it possible to get the colour of the X to match the colour of the other menu buttons? (for example, grey in the dark theme and blue in the MaterialDesign theme)

It's very likely that the X button wasn't adapted to different themes when first introduced. I can look into this one.

> In the "DarkBetter" theme, the triangle icon on the side menu which indicates whether sections are visible or folded gets bigger for folded sections. This does not happen in the "Dark" theme though.

This could be related to the image files themselves that are being used as the background images. I can look into this one, too.

> I was messing around in the right-click shortcut menu to see how it felt to navigate, and I accidentally removed a torrent because I touched "remove" when I meant to press "set channel". This could be a fatal "off-by-one error" if someone were to accidentally remove a whole label or tracker worth of torrents. In case anyone else has fat fingers like me, would it be possible to increase the vertical padding of the menu items on the mobile layout?

I'm thinking about increasing the font size of the entire page to 16px on mobile, as it's the recommended font size for mobile apps. That would also reduce the likability of those fatal off-by-one errors happening (I made an exactly same error of deleting a torrent myself.) This change, however, requires more fine tuning as it affects the entire page, so more tests need to be done before I push this one.

**Comment by jevenski** [comment]
### What would you like changed about the web client?

Removing the iPad plugin did leave a side-effect on iOS devices. A major regression is that the `contextmenu` event is not supported on iOS browsers [Source: MDN](https://developer.mozilla.org/en-US/docs/Web/API/Element/contextmenu_event#browser_compatibility), so we are unabled to bring up any context menus by long pressing.

### Describe the solution you'd like and propose possible alternatives.

I'm thinking about defining a custom event for iOS devices to respond to long pressing actions and bring up the context menus.
      
### Provide information and resources about the environment hosting the web client. 

Server: ruTorrent v5.0-Beta 2 on Ubuntu 22.04 LTS
Web Client: Latest Safari on iPhone Xs running iOS 17.6.1, and technically any browser on iOS because they all run with the same core on iOS.
Installation Method: zip source from version release

### Additional context

None.

**Comment by stickz** [comment]
@jevenski I've brought back the iPad plugin and only enabled it for Safari. This should resolve that problem. https://github.com/Novik/ruTorrent/commit/feac220cb12068e59cd008a4c48fe677d320efdd

**Comment by stickz** [comment]
@Airman8 @koblack @jevenski Thanks so much for your help! You've helped us eliminate 20 bugs before posting a stable release of ruTorrent! We've addressed your feedback! I've posted one final beta version of v5.0. Could you give it a try and give your thoughts about posting a stable version of ruTorrent? https://github.com/Novik/ruTorrent/releases/tag/v5.0-beta3

@jevenski and I would like to get started on `ruTorrent v5.1`. We can most certainly adjust font sizes, the torrent information layout, make sub menu improvements etc. in this minor release. My plan was to delay that until `ruTorrent v5.1`. We've made a lot of changes in `ruTorrent v5.0`. We need to start anther development cycle, to effectively resolve any potential problems.

**Comment by Airman8** [comment]
I looked at the themes again in v5.0-beta3

Side menu width is now consistent when I rotate to landscape orientation and then back to portrait again. On my Galaxy S9 the menu fills the whole viewport width, but I can navigate away with the X button so no worries.

Another very minor thing, maybe for a later version: The background colour of the X button seems to be getting affected by the backdrop fader while the side menu is open, so it looks darkened in the same way as the inactive buttons. In the Oblivion theme it's a bit low contrast, but it's still visible enough.

**Comment by koblack** [comment]
1. Oblivion - side panel on mobile

![image](https://github.com/user-attachments/assets/da283f53-f4d1-44ae-8f57-78f393b36e17)

2.  On all tabs cannot scroll content to the bottom. The content at the bottom cannot be displayed. In the Files tab, percentages are also displayed on the bottom bar.

![image](https://github.com/user-attachments/assets/548a1c9e-3860-46a6-9375-d72509e95ce3)
![image](https://github.com/user-attachments/assets/0b2b84c1-3a03-45a5-93b4-0a52fc36a0ad)

4. MaterialDesign / Oblivion - settings top right corner

![image](https://github.com/user-attachments/assets/34c85205-0441-4a4c-995f-efd1592add7e)
![image](https://github.com/user-attachments/assets/71571696-ff17-4c9e-bc1a-d2a427e06c46)

5. RSS Manager - maybe it's an intentional change, but the reset button changes its position

![image](https://github.com/user-attachments/assets/520b4cbd-cde0-46cb-bf39-b45d413b50fd)
![image](https://github.com/user-attachments/assets/b1b63b7c-af68-4525-a34c-136a2756321b)

6. just to check that it looks as it should

![image](https://github.com/user-attachments/assets/1d91c88a-9b55-40ff-a56b-2f43f31f5c28)
![image](https://github.com/user-attachments/assets/3662e98d-1429-457e-82de-40f54a38f8fd)

**Comment by jevenski** [comment]
Before: 
![Screenshot 2024-10-27 151043](https://github.com/user-attachments/assets/a70debc3-91af-499f-b97c-0e7bc7b73fe4)

After:
![Screenshot 2024-10-27 151805](https://github.com/user-attachments/assets/e1db2500-9332-45c5-9c0d-2ec4e66e3e15)


Grouped buttons overflow on mobile if there are too many of them, for example the logoff plugin would add a log-off button before the add torrent button, so there will be four buttons in total in this group if the create plugin is also in use. Submitting a patch to allow wrapping on mobile: https://github.com/Novik/ruTorrent/pull/2764

**Comment by jevenski** [comment]
### Steps to reproduce
1. Sort torrent by a column that's created by a plugin (e.g. seedingtime, upload eta, etc);
2. Disable the relevant plugin;
3. Refresh the page.

```javascript
Uncaught TypeError: Cannot read properties of undefined (reading 'getAttribute') from TypeError: Cannot read properties of undefined (reading 'getAttribute')
    at dxSTable.Sort (http://buchladen.local/js/stable.js?v=51B3:650:67)
    at dxSTable.syncDOM (http://buchladen.local/js/stable.js?v=51B3:1618:91)
    at http://buchladen.local/js/stable.js?v=51B3:1667:10
```

Submitting a patch to sort on "Name" column if the custom sort column is not present.

**Comment by jevenski** [comment]
### Steps to reproduce
1. Sort torrent by a column that's created by a plugin (e.g. seedingtime, upload eta, etc);
2. Disable the relevant plugin;
3. Refresh the page.

```javascript
Uncaught TypeError: Cannot read properties of undefined (reading 'getAttribute') from TypeError: Cannot read properties of undefined (reading 'getAttribute')
    at dxSTable.Sort (http://xxx/js/stable.js?v=51B3:650:67)
    at dxSTable.syncDOM (http://xxx/js/stable.js?v=51B3:1618:91)
    at http://xxx/js/stable.js?v=51B3:1667:10
```

Submitting a patch to sort on "Name" column if the custom sort column is not present.

**Comment by stickz** [comment]
@koblack @jevenski I would like to confirm that we're ready to tag a stable release of ruTorrent `v5.1.0`! If anything else comes up, I would be happy to post a hotfix. Is there anything else you'd like to do beforehand? 

This is looking very stable right now, so thanks for all your hard work testing!

**Comment by jevenski** [comment]
As far as I am aware of, there are still two things that needs to be adjusted.

1. Theme adaption for the plugins dropdown menu on the toolbar. It uses Bootstrap's default theme color now. I am planning to override some of Bootstrap's color variables (`--bs-dropdown-bg`, `--bs-dropdown-border-color`, etc.) in the `style.css` file of different themes, so future maintenance can be done in a finger snap: just redefine these variables in the theme's `style.css` file and everything is taken care of by the main `style.css` file.

2. The _getdir plugin returns to the upper directory if given a pat that exists on the server and is readable by the web-user, only without the trailing slash (e.g. `/home/user/download` instead of `/home/user/download/`). Currently, the front-end removes everything after the last slash and sends the remaining part to the server, so `/home/user/download` becomes `/home/user/`. If it's not readable or doesn't exist, the default top directory will be returned. 
For an easy fix, we can simply remove the "removing" part on the front-end. For a more sophisticated fix, we can further validate the given path, checking if it's a directory, a file or something inaccessible, and return the results respectively.

**Comment by stickz** [comment]
> As far as I am aware of, there are still two things that need to be adjusted.
> 
Okay, we can wait a little bit longer before tagging the stable release.

> 1. Theme adaption for the plugins dropdown menu on the toolbar. 
> 
Sounds like a plan.

> 2. The _getdir plugin... Currently, the front-end removes everything after the last slash and sends the remaining part to the server, so `/home/user/download` becomes `/home/user/`. 
>
We should be able to split the string (across `/`) then check the last part for a readable file. If not, we can check for directory and concatenate the forward slash. This is a very simplistic and unsophisticated algorithm.

**Comment by jevenski** [comment]
> We should be able to split the string (across `/`) then check the last part for a readable file. If not, we can check for directory and concatenate the forward slash. This is a very simplistic and unsophisticated algorithm.

The algorithm itself is not that complicated. The utility functions involved, however, need some adjustments.

https://github.com/Novik/ruTorrent/blob/5ff1dc8442f85330f4aa40246b2583178febdb17/php/utility/permission.php#L7-L28

This utility function returns `false` if given a readable file as the parameter, so the outer function `Permission::doesUserHave()` will always return `false` if given a non-directory parameter, whether it's a readable file or a non-readable path. I think this function should return `true` if the given parameter is a readable file or a readable directory. 

PHP has some built-in functions to check for user permission (`is_readable()`, `is_writable()`, `is_executable()`), and some others to check for directories (`is_dir()`, `is_file()`). We might be able to combine these functions to create new, simpler utility functions for checking user permissions. But it also seems that these functions runs under the Linux user that executes the PHP scripts, which can be `www-data` or `nobody`, but what we need is the permissions of the user running rtorrent, which can be different from the aforementioned user. I'm not an expert on PHP, though.

**Comment by koblack** [comment]
I don't see any significant issues. If I notice anything, I'll let you know.

**Comment by stickz** [comment]
> The algorithm itself is not that complicated. The utility functions involved, however, need some adjustments.
> 
It would be required to add new ones for instance `isUserHavePrimEx()`, but not until `v5.2`.

> I don't see any significant issues. If I notice anything, I'll let you know.
>
Okay, I'll wait a couple more days. If no issues turn up, I'll post `v5.1` stable.

**Comment by castillo92** [comment]
Hello, in general the new interface works fine for me, but have two main issues on mobile: Apple iOS 18.1

Selecting some folder and try to change it, it block the tree. For me it is necessary to refresh all page to reset the form.
![1](https://github.com/user-attachments/assets/a5c1da86-7fb7-40c1-ad00-6729bca32bff)

This is an improvement that I have been thinking about requesting for a long time: more space to show completely the file name selected. I selected the good file?
![2](https://github.com/user-attachments/assets/c0e26283-8373-4989-82d3-e68e097dfc3f)

**Comment by stickz** [comment]
Hello @castillo92 thanks so much for the feedback! We fixed number 1 and 3 already in the latest master branch. You can download the latest zip file here to test: https://github.com/Novik/ruTorrent/archive/refs/heads/master.zip

We have the iPad touch plugin enabled for Apple iOS, to bring up the context menu. It's not possible to solve at this point in time. This is a limitation of Apple iOS. It's required to keep this feature enabled to open the context menu.

I would like to hear @jevenski 's thoughts if we can put the `choose files` button beside `add files` buttons on mobile to double the room for the filename. Also, we need to increase the width of these buttons to allow for translations to fit properly on mobile!

**Comment by castillo92** [comment]
> Hello @castillo92 thanks so much for the feedback! We fixed number 1 and 3 already in the latest master branch. You can download the latest zip file here to test: https://github.com/Novik/ruTorrent/archive/refs/heads/master.zip
> 
> We have the iPad touch plugin enabled for Apple iOS, to bring up the context menu. It's not possible to solve at this point in time. This is a limitation of Apple iOS. It's required to keep this feature enabled to open the context menu.
> 
> I would like to hear @jevenski 's thoughts if we can put the `choose files` button beside `add files` buttons on mobile to double the room for the filename. Also, we need to increase the width of these buttons to allow for translations to fit properly on mobile!

Thanks you so much for the quick reply, I will try the latest master branch right now.
I edited and number the questions, could you please check again?

**Comment by stickz** [comment]
It looks like we have another issue where the x button doesn't close the directory listing on mobile as well.

We can't mark `v5.1` as stable until this is resolved.

**Comment by lottelucky** [comment]
The plugin DUMP cannot be started. An error in the log: The dumptorrent program cannot be called. How should I configure this plugin?

**Comment by jevenski** [comment]
Submitting a pull request for this one: https://github.com/Novik/ruTorrent/pull/2769. 

Browsers have their own styles for file inputs. It requires a bit adjustments and I think could be delayed for a later release. For a simple fix, we can add another text box beside the add file button to display the complete file names.

**Comment by stickz** [comment]
@castillo92 Could you install `master.zip` again to confirm a fix for the scrolling and x button issues on IOS? I've merged the pull request created to resolve the problem. We will wait a confirmation before tagging the stable release.
https://github.com/Novik/ruTorrent/archive/refs/heads/master.zip

I agree with @jevenski about the the file name for "add torrent". We are going to set a milestone of `v5.2` to resolve this issue. It will require a significant change, which will setback our release timeline of `v5.1`. The scope of the issue is relatively minor.

@lottelucky It's required to build this software manually with GNU make and a compiler like GCC. It's recommended to submit an issue report to the docker container. If they would like the contribution, I will submit a pull request. You can delete or disable this plugin for now. The functionality is not required. https://github.com/mondediefr/docker-rutorrent

**Comment by jevenski** [comment]
### Steps to reproduce

1. Scroll an s-table horizontally to some point;
2. Drag a header cell;
3. The drop target box (whatever it is called) wouldn't appear.
![Screen Recording 2024-10-29 181903](https://github.com/user-attachments/assets/9719c63e-b8ab-4c15-b005-215dcee63469)
4. Scroll the table to the left-end would produce the expected effect for all scenarios.
![Screen Recording 2024-10-29 182013](https://github.com/user-attachments/assets/6d9a13ef-6cc1-4fd0-8f03-380d9e1216c5)

Submitting a patch: https://github.com/Novik/ruTorrent/pull/2770

**Comment by jevenski** [comment]
> I unzipped the two files "dumptorrent" and "scrapec" into the directory "/rutorrent/app/plugins/dump" and gave it executable permissions with the chmod + x command, but restarting rutorrent still failed to start the dump plugin.

`rutorrent/app/plugins/dump/` should not be a system search path for executable binaries, so it needs to be set in the `conf.php` file:
```PHP
$pathToExternals['dumptorrent'] = '';
```

**Comment by koblack** [comment]
https://github.com/Novik/ruTorrent/issues/1866 
This one old issue annoying me from time to time.
@jevenski Can you reproduce this issue?

**Comment by jevenski** [comment]
> https://github.com/Novik/ruTorrent/issues/1866 
> 
> This one old issue annoying me from time to time.
> 
> @jevenski Can you reproduce this issue?

I have been through this situation before. Now I add torrents mostly by adding URLs, so I haven't seen it for some time. Another reason is that I do less torrenting these years. Owner said this was a Firefox issue, and couldn't be fixed at APP level.

**Comment by castillo92** [comment]
> @castillo92 Could you install `master.zip` again to confirm a fix for the scrolling and x button issues on IOS? I've merged the pull request created to resolve the problem. We will wait a confirmation before tagging the stable release. https://github.com/Novik/ruTorrent/archive/refs/heads/master.zip
> 
> I agree with @jevenski about the the file name for "add torrent". We are going to set a milestone of `v5.2` to resolve this issue. It will require a significant change, which will setback our release timeline of `v5.1`. The scope of the issue is relatively minor.
> 
> @lottelucky It's required to build this software manually with GNU make and a compiler like GCC. It's recommended to submit an issue report to the docker container. If they would like the contribution, I will submit a pull request. You can delete or disable this plugin for now. The functionality is not required. https://github.com/mondediefr/docker-rutorrent

Hello @stickz !

I checked the latest master branch and yes: button [X] in "Add torrent" menu works as expected, using it, changing path, adding different torrents and scrolling through torrent list is now very smooth on iOS 18.1 (latest at this moment)

I checked change the list of "limit download/upload speed" and also works fine

**Comment by stickz** [comment]
Great job testing everyone! Keep up the good work! We are getting very close to tagging the stable `v5.1` release!

**Comment by jevenski** [comment]
> ![Screenshot_20241021_091054_Chrome](https://github.com/user-attachments/assets/64d45972-2827-47b6-93f7-013b222f1f19)

_Originally posted by @koblack in https://github.com/Novik/ruTorrent/issues/2748#issuecomment-2425825286_
            
Screenshots on mobile has overflow flaws. Submitting a PR to address this issue: https://github.com/Novik/ruTorrent/pull/2773

**Comment by stickz** [comment]
@koblack @jevenski @castillo92 I've posted **ruTorrent v5.1-beta4**. Your help testing would be greatly appreciated. I've decided to delay **ruTorrent v5.1 stable** by 1 week to test more. https://github.com/Novik/ruTorrent/releases/tag/v5.1-beta4

**Comment by Airman8** [comment]
I can't vertically scroll the RSS Manager on mobile, so everything below the heading "RatioGroup" is invisible. This includes the RatioGroup selection box, the Channel heading and selection box, as well as the Reset, OK, and Cancel buttons.

Very minor UI thing I also noticed with the left menu. The X button is shaded to look like it's in the background even though it's in the foreground.

![IMG_20241103_223604](https://github.com/user-attachments/assets/52593519-f866-423d-ab7b-fdc8495ee078)

Other than that, I haven't noticed any major issues.

**Comment by adapowers** [comment]
Hi! Love this feedback thread, and love the 5.1 beta. Some enhancements that would go a long way for me on mobile: (Safari, iPhone 14 Pro)
- +1 to the long-press issues; to avoid activating context menu on a row, I basically have to tap and flick, which also makes me run into:
- The elasticity on the table is set to infinite, so while scrolling you constantly have to tap to “pin it down”.
- No way to multi-select rows! This is currently the biggest pain point for me.
- Adjustable row sizing. It’s currently desktop-parity, but having the option to make the rows chunkier would be so great.

Thank you!

**Comment by koblack** [comment]
Side panel comparison v4.3 (left) vs v 5.1 (right)
Minor issues with MaterialDesign and Oblivion

* MaterialDesign

![MaterialDesign5](https://github.com/user-attachments/assets/b7116cb7-b5a3-4221-a51b-e98af9282230)

* Oblivion

![Oblivion3](https://github.com/user-attachments/assets/d3e8b6c0-4831-4a90-b3a3-07e2aa33e999)

* Acid

![Acid](https://github.com/user-attachments/assets/1439e725-461c-4bc5-9930-a21eb2580fa4)

* Blue

![Blue](https://github.com/user-attachments/assets/29e6afd6-4750-4e91-afe6-197271263da7)

* DarkBetter

![DarkBetter](https://github.com/user-attachments/assets/286bc524-4ae4-40fd-9214-e38067fc283d)

* Excel

![Excel](https://github.com/user-attachments/assets/0e9e9328-7c95-47c0-bfae-62d054041e63)

* Standard

![Standard](https://github.com/user-attachments/assets/aca915f8-d46a-486b-939e-574d9d85a2df)

**Comment by koblack** [comment]
Menu comparison v4.3 (top) vs v 5.1 (bottom)
There are some color changes here and there, Oblivion and DarkBetter most significantly.

* MaterialDesign

![MaterialDesign](https://github.com/user-attachments/assets/f052dded-cd3e-4e41-b86f-cbb4ce628838)

* Oblivion

![Oblivion](https://github.com/user-attachments/assets/31377524-93f1-4390-8b42-7dba587efe0a)

* Acid

![Acid](https://github.com/user-attachments/assets/19137e4d-a3b3-4bb1-a0eb-45f6369a45e3)

* Blue

![Blue](https://github.com/user-attachments/assets/be2877a4-b3fb-43c3-934a-2ca41c44f55f)

* DarkBetter

![DarkBetter](https://github.com/user-attachments/assets/bfd7093b-ed85-400f-8646-1625404b80fe)

* Excel

![Excel](https://github.com/user-attachments/assets/c220e3fd-90f7-4fc0-903b-15759b27fb2e)

* Standard

![Standard](https://github.com/user-attachments/assets/5c7d7ca8-58da-4989-a0f2-c2d0f2429459)

**Comment by koblack** [comment]
General tab comparison v4.3 (left) vs v 5.1 (right)

* MaterialDesign

![MaterialDesign](https://github.com/user-attachments/assets/488c0fb5-d00d-4896-914c-10f549732a86)

* Oblivion

![Oblivion](https://github.com/user-attachments/assets/86aaf505-33c7-492b-8350-5cebb158147b)

* Acid

![Acid](https://github.com/user-attachments/assets/317dac6a-41d0-418e-a931-013e7bf273e7)

* Blue

![Blue](https://github.com/user-attachments/assets/e8f2c7f3-1106-4dcb-a65c-d3819bc289cc)

* DarkBetter

![DarkBetter](https://github.com/user-attachments/assets/6ba03d10-752e-45ba-87e6-4509839d26ff)

* Excel

![Excel](https://github.com/user-attachments/assets/113a2531-43e8-411b-be82-743a47674b8d)

* Standard

![Standard](https://github.com/user-attachments/assets/0c67212d-4448-4db2-ab1c-0f161194482c)

**Comment by jevenski** [comment]
> I can't vertically scroll the RSS Manager on mobile, so everything below the heading "RatioGroup" is invisible. This includes the RatioGroup selection box, the Channel heading and selection box, as well as the Reset, OK, and Cancel buttons.

Submitting a PR for this one: https://github.com/Novik/ruTorrent/pull/2778. It was previously planned for v5.2, but now it seems more urgent that we'd better include this in v5.1 stable.

**Comment by stickz** [comment]
> > I can't vertically scroll the RSS Manager on mobile, so everything below the heading "RatioGroup" is invisible. This includes the RatioGroup selection box, the Channel heading and selection box, as well as the Reset, OK, and Cancel buttons.
> 
> Submitting a PR for this one: #2778. It was previously planned for v5.2, but now it seems more urgent that we'd better include this in v5.1 stable.

@Airman8 Thanks so much for reporting this issue! We've now resolved it! Could you confirm a successful resolution by trying the latest master zip file? https://github.com/Novik/ruTorrent/archive/refs/heads/master.zip

**Comment by stickz** [comment]
> Hi! Love this feedback thread, and love the 5.1 beta. Some enhancements that would go a long way for me on mobile: (Safari, iPhone 14 Pro)
> 
> * +1 to the long-press issues; to avoid activating context menu on a row, I basically have to tap and flick, which also makes me run into:
> * The elasticity on the table is set to infinite, so while scrolling you constantly have to tap to “catch it”.
> * No way to multi-select rows! This is currently the biggest pain point for me.
> * Adjustable row sizing. It’s currently desktop-parity, but having the option to make the rows chunkier would be so great.
> 
> Thank you!

Thanks for the feedback @adapowers. Could you try downloading the Google Chrome application on your iPhone 14 Pro and report back how much easier it is to operate ruTorrent? Safari has known item selection issues, which can't be addressed in `v5.1`.

Also, we plan to make more improvements like font and row sizing in `v5.2`.

**Comment by castillo92** [comment]
Hello @stickz

A little UI issue, for me low priority and importance. If you try to add a torrent and press [X] button to close the window, lot of times is also pressed "Create torrent" button behind it.

https://github.com/user-attachments/assets/8602a3ad-4c1a-43af-a40a-a279658ed875


And also low priority because works but not as i liked, the priority menu is popped outside-left screen if you click on a file inside a multiple file torrent, preventing user to read the list and do not know what are you clicking.

https://github.com/user-attachments/assets/8ee40853-7e35-4c7a-98af-06ce78c9f355

**Comment by Airman8** [comment]
> @Airman8 Thanks so much for reporting this issue! We've now resolved it! Could you confirm a successful resolution by trying the latest master zip file? https://github.com/Novik/ruTorrent/archive/refs/heads/master.zip

@stickz We're closer to resolution, but there's one remaining issue. The content of the dialogue box scrolls properly, but the position of the three buttons is fixed and partially hidden behind the soft keys of my phone's UI.

Phone is a Samsung Galaxy S9 running LineageOS (old and completely lacking a back plate, but still hanging on).

This is as much as I can see of the buttons:

![IMG_20241105_215600](https://github.com/user-attachments/assets/2c4c6109-44a4-4b68-a9cd-dcbbeb022d60)

**Comment by gijsyo** [comment]
I am now running 5.1.B4 and I've noticed that the diskspace graphic and the port status indicater images do not load. I did recently switch from single user to multi user though. Any pointers as to where I might start to look? I cleared my browsers cache but to no avail. AFAIK, Apache isn't caching anything.

**Comment by jevenski** [comment]
> ![IMG_20241105_215600](https://private-user-images.githubusercontent.com/16048904/383177396-2c4c6109-44a4-4b68-a9cd-dcbbeb022d60.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEwNjQ5NTEsIm5iZiI6MTczMTA2NDY1MSwicGF0aCI6Ii8xNjA0ODkwNC8zODMxNzczOTYtMmM0YzYxMDktNDRhNC00YjY4LWE5Y2QtZGNiYmViMDIyZDYwLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA4VDExMTczMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVmNzAwMTRhNTAxYTYxZGU1YTYzOWEwZTJkMDJkZWM1NDEzMmEwYTEyYmYyNmU5Y2Q2OGExYWVkMzNjM2E1ZmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.FjnwT4-nf3663PjkURJiFTOn8gJmvjNPGJSq40VtMTk)

https://github.com/Novik/ruTorrent/blob/a1c79696154c4d9561485ff1f6859b77285df0a0/css/style.css#L149-L150

I'm thinking the dynamic viewport height could be the culprit. The dialog window now takes 95% of viewport height which is what the `95vh` rule do, and the bottom part of the viewport is taken by the system navigation bar, and hence the overlapping. Changing the units of `vh` and `vw` to `dvh` and `dvw` in these lines might solve the problem, like we did for the height of the main content. Judging from the screenshot, the height of the main content and the position of the status bar is correct, and we have already applied `dvh` rules on that part.
However, I don't have such a device to verify the solution. The mobile interface on a desktop browser's DevTools always has a fixed viewport height. My android phone doesn't have such a display issue. Maybe you could do such a modification on your server and see how it turns out, rather than I submitting an untested PR? If it does solve the problem, you can file a PR and become a contributor yourself!

**Comment by koblack** [comment]
Quick test of both units. It seems that `dvh` fixing the problem. Screenshots below.

```
  max-width: 95vw;
  max-height: 95vh;
```

![Screenshot_20241108_200327_Chrome](https://github.com/user-attachments/assets/1f4a060a-5ea9-4663-8c5c-fa7ef3c11b80)
![Screenshot_20241108_200307_Chrome](https://github.com/user-attachments/assets/a02c3818-fb2b-4cbd-9df5-de4f5fc16f76)

```
  max-width: 95dvw;
  max-height: 95dvh;
```

![Screenshot_20241108_200646_Chrome](https://github.com/user-attachments/assets/247ea2dc-51b4-4dbc-95c6-3a99c5d39b97)
![Screenshot_20241108_200609_Chrome](https://github.com/user-attachments/assets/a25a62de-4862-4614-adea-6804b95b9446)

**Comment by Airman8** [comment]
> Quick test of both units. It seems that `dvh` fixing the problem. Screenshots below.

This fix worked for me as well, with the same results as @koblack had.

![Screenshot_20241109-035632_Firefox](https://github.com/user-attachments/assets/655201d6-90a1-4f9f-9456-07f3cf75b49e)

I tested the other windows as well, and nothing changed from before, so they all work still too.

> If it does solve the problem, you can file a PR and become a contributor yourself!

@jevenski Sure thing! I have submitted a PR for this fix.

**Comment by stickz** [comment]
We've released Beta 5 to address everyone's feedback! Are we ready now for the stable release?
https://github.com/Novik/ruTorrent/releases/tag/v5.1-beta5

**Comment by Airman8** [comment]
The issue with the RSS downloader remains resolved in Beta 5, and I've found no other issues so far.

**Comment by jevenski** [comment]
> Hello Stickz, testing 5.1B5. At first sight, no bugs.
> 
> 
> 
> But I ask you a question... I allways make an update of ruTorrent, I lost the list of limits od DL an UP Speeds.
> 
> 
> 
> Where is stored to make it backupeable?
> 

User settings and uploaded torrent files (if set to be saved) are stored in `share/`. You should copy this folder to the new installation as well as the `conf/` folder.

**Comment by ranirahn** [comment]
I found very interesting bug and I dont even understand how is this possible.
What i did
Add Torrent->Select directory (example NEW MOVIES)->paste magnet link to Torrent URL box->Click Add URL

Now I have 2 folders named NEW MOVIES. I did not think i even could greate 2 folders with exactly the same name. I am so confused. Now old movies are in old folder and new movies are in new folder.

**Comment by koblack** [comment]
Maybe it's worth starting a new thread regarding the v5.2 release?

**Comment by stickz** [comment]
> Maybe it's worth starting a new thread regarding the v5.2 release?

File separate issue reports for now. I'll close this off and we'll create a new thread when `v5.2` goes to beta.

**Comment by TMTYD** [comment]
Hi, I have the latest version of rutorrent, but some areas in the Oblivion theme are still displayed in the wrong color.

<img width="1700" height="1044" alt="Image" src="https://github.com/user-attachments/assets/60297f2c-908f-42c5-b7c6-266e92e74d9b" />
<img width="753" height="662" alt="Image" src="https://github.com/user-attachments/assets/fff8071a-88a7-42b2-9c8c-9ec9e6b3913a" />
<img width="461" height="24" alt="Image" src="https://github.com/user-attachments/assets/7e7bdc23-ab5c-4c6a-a35f-d01f9a3295cc" />

<img width="573" height="95" alt="Image" src="https://github.com/user-attachments/assets/04e5dbf6-c9b6-431d-b440-2c503eefb3c0" />
<img width="896" height="137" alt="Image" src="https://github.com/user-attachments/assets/dd810122-991d-44dc-8003-b88408667605" />
<img width="924" height="85" alt="Image" src="https://github.com/user-attachments/assets/cd0a2c90-b230-450d-9850-9d7f2f2e1e12" />
<img width="1051" height="693" alt="Image" src="https://github.com/user-attachments/assets/32d7ecf7-7961-4179-b2d6-96d12e49ba70" />
